src/sharing/SharingDialog.component.js

```diff
@@ -1,44 +1,251 @@
-import { config } from 'd2/lib/d2';
+/* eslint no-console: 0 */
+/* eslint class-methods-use-this: 0 */
+/* eslint react/jsx-no-bind: 0 */
+/* eslint no-unused-expressions: 0 */
+
+import { config, getInstance } from 'd2/lib/d2';
 import Dialog from 'material-ui/Dialog/Dialog';
 import FlatButton from 'material-ui/FlatButton/FlatButton';
-import React, { PropTypes, createClass } from 'react';
-import Translate from '../i18n/Translate.mixin';
+import React, { PropTypes } from 'react';
 import Sharing from './Sharing.component';
-import sharingStore from './sharing.store';
+import LoadingMask from '../loading-mask/LoadingMask.component';
 
+config.i18n.strings.add('share');
 config.i18n.strings.add('close');
-config.i18n.strings.add('sharing_settings');
 
-export default createClass({
-    propTypes: {
-        objectToShare: PropTypes.object.isRequired,
-        onRequestClose: PropTypes.func.isRequired,
-    },
+function cachedAccessTypeToString(canView, canEdit) {
+    if (canView) {
+        return canEdit
+          ? 'rw------'
+          : 'r-------';
+    }
+
+    return '--------';
+}
+
+function transformAccessObject(access, type) {
+    return {
+        id: access.id,
+        name: access.name,
+        displayName: access.displayName,
+        type,
+        canView: access.access && access.access.includes('r'),
+        canEdit: access.access && access.access.includes('rw'),
+    };
+}
+
+function transformObjectStructure(apiMeta, apiObject) {
+    const userGroupAccesses = !apiObject.userGroupAccesses ? [] : apiObject.userGroupAccesses.map(
+        access => transformAccessObject(access, 'userGroup'));
+
+    const userAccesses = !apiObject.userAccesses ? [] : apiObject.userAccesses.map(
+        access => transformAccessObject(access, 'user'));
+
+    const combinedAccesses = userGroupAccesses.concat(userAccesses);
+
+    return {
+        ...apiObject.user && {
```

Comment:
This construct is somewhat confusing. 

It will leave out the `authorOfSharableItem` when `apiObject.user` is `undefined` or `null` is that the intended result or is it primarily to guard against `apiObject.user` to be falsy?